harwest/lib/atcoder/workflow.py

```diff
@@ -0,0 +1,98 @@
+import os
+from harwest.lib.utils import config
+
+from harwest.lib.atcoder.client import AtcoderClient
+from harwest.lib.utils.repository import Repository
+from harwest.lib.utils.submissions import Submissions
+
+
+class AtcoderWorkflow:
```

Comment:
Would completely agree with you that the behavior of this workflow is rather similar to that of the Codeforces one with some more minor changes in the flow leading to quite a bit of duplication of code. How about we extract out the common behavior into a separate file `lib/AbstractWorklfow.py` and make both the Codeforces and AtCoder workflow extend it?

Reply:
yeah, i was thinking about that as well, it'll make it a bit easier to manage as well